seqr/views/apis/dataset_api.py

```diff
@@ -41,48 +42,45 @@ def add_dataset_handler(request, project_guid):
     project = get_project_and_check_permissions(project_guid, request.user, permission_level=CAN_EDIT)
 
     request_json = json.loads(request.body)
-    if 'form' not in request_json:
-        return create_json_response(
-            {}, status=400, reason="Invalid request: 'form' not in request_json")
 
-    form_data = request_json['form']
+    logger.info("add_dataset_handler: received %s" % pformat(request_json))
 
-    logger.info("add_dataset_handler: received %s" % pformat(form_data))
-
-    if "sampleType" not in form_data or "datasetType" not in form_data or "genomeVersion" not in form_data or "datasetPath" not in form_data:
+    required_fields = ['sampleType', 'datasetType']
```

Comment:
why isn't elasticsearchIndex required?

Reply:
for adding bam/cram datasets theres no index